nearlib/dist/nearlib.js

```diff
@@ -260,16 +291,24 @@ class Near {
             contract[methodName] = async function (args) {
                 args = args || {};
                 const response = await near.scheduleFunctionCall(0, options.sender, contractAccountId, methodName, args);
-                let status;
+                let result;
                 for (let i = 0; i < MAX_STATUS_POLL_ATTEMPTS; i++) {
                     await sleep(STATUS_POLL_PERIOD_MS);
-                    status = await near.getTransactionStatus(response.hash);
-                    if (status.status == 'Completed') {
-                        return status;
+                    result = (await near.getTransactionStatus(response.hash)).result;
+                    const flatLog = result.logs.reduce((acc, it) => acc.concat(it.lines), []);
+                    flatLog.forEach(line => {
+                        console.log(`[${contractAccountId}]: ${line}`);
```

Comment:
I would assume something like 
`sdf3`: log1
`sdf3`: log2 
`sdf3` -> `bcd3`: log3

where `bcd3` is receipt hash from `sdf3` tx. 4 letters are enough too.

Reply:
let's get back to this when hashes are represented as strings